### PR TITLE
Add testing/readme and image

### PR DESCRIPTION
**Pull request by r-rojo**

### CODE REVIEW

File: testing/readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+![OPAE mock test](mock_driver.png "OPAE mock test") 
+
+### test_system ###
+
+As mentioned above, `test_system` is a singleton class that is designed to be
+called from either C code (by using its C interface) or by using its C++
+`instance()` function. This allows it to be called from both google test tests
+as well as from C functions that are being mocked. Tests that depend on any
+sysfs paths handled by the OPAE kernel driver should be implemented as part of
+a test fixture. This fixture should call `test_system::initialzie` in its setup
```
Comment (tswhison) [inline testing/readme.md]: s/initialzie/initialize/

File: testing/readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# OPAE Testing Framework #
+
+## Objective ##
+
+The main goal of the testing framework used by OPAE is to enable (Google test)
+tests that don't require the presence of either an OPAE kernel driver or an
+OPAE compatible FPGA. An additional goal is to allow authoring tests that can
+easily ported from a mock environment to a real system.
```
Comment (tswhison) [inline testing/readme.md]: Is porting strictly required to run these on hardware, or is it a matter of re-linking?

File: testing/readme.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+The two interfaces to the OPAE Linux driver are `ioctl` calls and `sysfs`
+files. The OPAE C library uses both of these interfaces in its implementation
+to control and/or query the device. To test the API functions (including
+internal functions used within), a `mock` driver has been developed that can
+allow for calling these functions without calling the kernel driver through
+these interfaces. At a high level, system calls related to `ioctl` and `sysfs`
+files (eg. `ioctl`, `open`) are implemented in a C file called `mock.c` which
+in turn calls into a C++ singleton class called `test_system` using its C
+interface. This C++ singleton can also be used the any google test code to call
```
Comment (ibarraza) [inline testing/readme.md]: Typo.

File: testing/readme.md (hunk 1 above)
Comment (ibarraza) [inline testing/readme.md]: Typo.

File: testing/readme.md (hunk 2 above)
Comment (r-rojo) [reply testing/readme.md]: I think you could run most (if not all) of these tests on real HW by omitting the mock.c file and making sure the paths aren't re-rooted.

**Pull request merged**